### DISCUSSION TITLE
Page jumps to the top when the page is refreshed via htmx

### DISCUSSION THREAD

**Discussion by yatesco** [bigskysoftware/htmx#1009] [Q&A]
(I'm *sure* it never used to, so I'm questioning my sanity ;-)).

I have a page with a bunch of elements including textareas. The textareas POST when they lose focus and the reponse is a full page which re-renders outerHTML of the root element, e.g. a full page "refresh".

However, after the page refresh the position of the page is at the top, even if before that refresh the page had been scrolled down.

Is this expected default behaviour, and how do I stop it? I haven't overridden any globals and I'm not setting a scroll target in `hx-swap`. I don't want to jump to a known ID either, I simply want to scroll position remembered before the refresh and reapplied after the refresh. I could write some JS myself listening to the relevant events, but I'm absolutely convinced it didn't do this before. 

Help :-)

**Comment by jreviews** [comment]
I feel like this may be related to the introduction of the `focus-scroll` swap modifier and `htxm.config.defaultFocusScroll` global config. More info here https://htmx.org/attributes/hx-swap/#focus-scroll and here https://htmx.org/docs/#config

Can you try setting the global `htxm.config.defaultFocusScroll` to `true`?

These were introduced to prevent htmx from scrolling away when you are filling out a form that works in a similar way as yours, and you've already scrolled to a different position before the previous response is swapped. Like scrolling down to submit a long form using a button and then suddenly have the page jump up before you are able to submit it.

If your inputs don't have IDs, then it's also possible it's just a matter of adding the ID attributes without necessarily including it in your `hx-swap`.

**Comment by yatesco** [comment]
I tried that by tying ‘ htxm.config.defaultFocusScroll = true’ but the behaviour is the same

**Reply by jreviews** [reply]
Do the textareas have id attributes?

**Comment by yatesco** [comment]
I tried that as well, but same behaviour

**Comment by jreviews** [comment]
Can you create an example that reproduces the error using https://htmx.org/docs/#creating-demos and try rolling back htmx through different versions until you figure out on which version the issue started happening?

**Reply by yatesco** [reply]
well, now I'm confused. I haven't set up a demo yet, I'm just playing locally, but I've basically got a page which returns 100 `textarea`s with the same config as I'm using, and it works fine - no scrolling or anything.

I guess my actual question is: what on earth have I done which causes htmx to jump to the top after refreshing :-)

**Reply by jreviews** [reply]
Well, at least now you know. Maybe check for javascript errors in the browser console?

**Reply by yatesco** [reply]
nope - nothing :-(.

**Reply by yatesco** [reply]
Found a reproducible case. The problem is when I have a div with a size and `overflow:auto`. The scroll position *within that div* is lost.

Shall I put together a reproducible demo?

**Reply by jreviews** [reply]
Maybe @1cg can provide some feedback here on whether this is something workable. I don't think htmx can maintain the scroll position within a child element.

**Reply by yatesco** [reply]
I think one significant difference is that I am replacing the *entire page* on each refresh. I can't reproduce it using the mock POST/templates demo approach because it is always a sibling that gets replaced, and that works, even with `overflow: auto` on the sibling.

Trying to nail this down further....

**Reply by yatesco** [reply]
GOT IT :-) - sort of. So I have a top level div which gets replaced using:
```html
<div class="my-class" hx-target=".my-class">
  <... style="width:200px;overflow:auto;">
    <textarea hx-post="/whatever"/>
  <...>
</div>
```
this replaces the HTML *but* loses the scroll position. Simply changing it from a class to an ID and everything works smoothly:
```html
<div id="my-class" hx-target="#my-class">
  <... style="width:200px;overflow:auto;">
    <textarea hx-post="/whatever"/>
  <...>
</div>
```

Also note that this only works for one level. It seems that the element that `overflow`s needs an id on itself and every ancestor, but I can't be definitive. Sticking ids everywhere has made it work.

**Reply by yatesco** [reply]
Nope - it still doesn't work :-(

**Comment by yatesco** [comment]
Will do

**Comment by yatesco** [comment]
Full disclosure - I've started a higher level meta-thread on https://github.com/bigskysoftware/htmx/discussions/1017

**Comment by gerbil** [comment]
Is it even possible to keep scroll for refreshed sections? like tables for instance
